src/test/kotlin/it/pagopa/wallet/services/WalletServiceTest.kt

```diff
@@ -0,0 +1,59 @@
+package it.pagopa.wallet.services
+
+import it.pagopa.wallet.WalletTestUtils
+import it.pagopa.wallet.client.NpgClient
+import it.pagopa.wallet.exception.BadGatewayException
+import it.pagopa.wallet.repositories.WalletRepository
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.any
+import org.mockito.kotlin.given
+import org.mockito.kotlin.mock
+import reactor.core.publisher.Mono
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class WalletServiceTest {
+    private val walletRepository: WalletRepository = mock()
+
+    private val npgClient: NpgClient = mock()
+
+    private val walletService: WalletService = WalletService(walletRepository, npgClient)
+    @Test
+    fun `createWallet creates wallet successfully`() = runTest {
+        /* preconditions */
+        val expected = Pair(WalletTestUtils.VALID_WALLET, WalletTestUtils.GATEWAY_REDIRECT_URL)
+
+        given(walletRepository.save(any())).willReturn(Mono.just(expected.first))
+        given(npgClient.orderHpp(any(), any())).willReturn(Mono.just(WalletTestUtils.hppResponse()))
+
+        /* test */
+        val actual = walletService.createWallet()
+
+        /* assertions */
+        assertEquals(expected, actual)
+    }
+
+    @Test
```

Comment:
Add tests to cover missing returnUrl and securityToken in the received NPGResponse

Reply:
Solved in [06962c6](https://github.com/pagopa/pagopa-wallet-service/pull/10/commits/06962c6879ab1fe84fffe90517ccc90bcd84dcc7)